compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -2206,8 +2207,9 @@ object Parsers {
         isOperator = !(location.inArgs && followingIsVararg()),
         maybePostfix = true)
 
-    /** PrefixExpr   ::= [`-' | `+' | `~' | `!'] SimpleExpr
-    */
+    /** PrefixExpr       ::= [PrefixOperator'] SimpleExpr
+     *  PrefixOperator   ::=  ‘-’ | ‘+’ | ‘~’ | ‘!’
```

Comment:
`+`, `-` and `~` are not working.

```scala
class Foo(var value: Int):
  def `unary_+_=`(value: Int): Unit = this.value = +value
  def `unary_-_=`(value: Int): Unit = this.value = -value
  def `unary_~_=`(value: Int): Unit = this.value = ~value
end Foo

def test =
  val x = Foo(9)
  +x = 10 // error: value unary_+ is not a member of Foo - did you mean x.unary_+_=?
  -x = 10 // error: value unary_- is not a member of Foo, but could be made available as an extension method.
  ~x = 10 // error: value unary_~ is not a member of Foo - did you mean x.unary_~_=?
```

Reply:
That's because Foo is missing `unary_+` and so on. So that's as expected.

Reply:
Does that mean that we cannot write `!x = 3` if we have `unary_!_=` but not `unary_!`?

Reply:
Yes, exactly

Reply:
Why? Do we typecheck `x.unary_!` in `x.unary_! = 3` before fully desugaring it into `x.unarry_!_=(3)`?

Reply:
Something like that, yes. Using a setter `foo_=` as `x.foo = y` is only allowed/attempted when `x.foo` exists itself. This is the same in Scala 2, and I think it's mentioned in the spec somewhere.